Final/TCPServer.cpp

```diff
@@ -0,0 +1,113 @@
+#include "TCPServer.h"
+
+#include "ThreadPool.h"
+#include "LinkedList.h"
+
+// Create linked list
+TCPServer::TCPServer(size_t max_concurrency)
+{
+    thread_pool = new ThreadPool(max_concurrency);
+    open_conns = new LinkedList();
+}
+
+// Clean up nodes
+TCPServer::~TCPServer()
+{
+    close(sock_fd);
```

Comment:
note, you'll also need to `free` the `thread_pool` and `open_conns` here. The `new` syntax in c++ is essentially a wrapper around the following in c (note that C doesn't really have the concept of classes or constructors, so it's a loose comparison i'm making):

```
open_conns = (LinkedList *)malloc(sizeof(struct LinkedList)
```

But the point is, using `new` will allocate memory in the heap that will need to be free'd.